server/src/main/java/com/thoughtworks/go/server/service/PluginService.java

```diff
@@ -70,7 +75,7 @@ public PluginSettings getPluginSettings(String pluginId) {
         if (plugin instanceof NullPlugin) {
             return null;
         } else {
-            pluginSettings.populateSettingsMap(plugin);
+            pluginSettings.populateSettingsMap(plugin, pluginInfoFor(pluginId));
         }
```

Comment:
'getPluginSettingsFor' and 'getPluginSettings' seem to be doing the same thing, any idea why we have both?

Reply:
`getPluginSettingsFor` is called from 

https://github.com/gocd/gocd/blob/fcf7fd9a32bc47b94c2dc6a23de845e6155fe561/server/webapp/WEB-INF/rails.new/app/controllers/admin/plugins/plugins_controller.rb#L51-L54

And

https://github.com/gocd/gocd/blob/fcf7fd9a32bc47b94c2dc6a23de845e6155fe561/server/webapp/WEB-INF/rails.new/app/controllers/admin/plugins/plugins_controller.rb#L56-L59

We can remove `getPluginSettingsFor(String pluginId, Map<String, String> parameterMap)` and ` getPluginSettingsFor(String pluginId)` with old plugin settings page.